src/pages/_document.tsx

```diff
@@ -0,0 +1,68 @@
+import { Children } from 'react';
+import { ServerStyleSheets } from '@material-ui/core/styles';
+import theme from '../theme';
+import Document, { Head, Html, Main, NextScript } from 'next/document';
+
+export default class MyDocument extends Document {
```

Comment:
It would be great if you could walk me through this, as I've never used `_document` before and am not sure why it's necessary for MUI.